### PR TITLE
Support numhl highlights for vim >= 8.2.3874

### LINKED ISSUE #4579: numhl highlights for vim isn't supported by ALE
In #2637, support for numhl highlights was added for nvim.

In the meantime, vim added support for numhl highlights in patch 8.2.3874.

However, ALE still doesn't allow numhl highlights to be enabled in vim.

Fix in #4578.

### PR DESCRIPTION
**Pull request by alt-romes**
Fixes #4579

In #2637, support for numhl highlights was added for nvim.

In the meantime, vim added support for numhl highlights in patch 8.2.3874.

This patch allows numhl highlights to be enabled in ALE for vim >= 8.2.3874 too.

### CODE REVIEW

**Comment by alt-romes** [comment]
Fixes #4579

**Review by pbnj-dragon — APPROVED** [review]
Looks good to me.

**Comment by alt-romes** [comment]
Is there anything left to do to @pbnj-dragon ?

**Comment by pbnj-dragon** [comment]
I don't have merge rights.
I defer to project maintainers: @w0rp or @hsanson

**Review by w0rp — APPROVED** [review]

**Comment by w0rp** [comment]
Cheers! :beers:

**Comment by alt-romes** [comment]
Thanks @w0rp !

**Pull request merged**
